packages/react/src/screens/auth-blocks/passkey-append/EditUserData.tsx

```diff
@@ -0,0 +1,76 @@
+import type { PasskeyAppendBlock } from '@corbado/shared-ui';
+import type { FC } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { PrimaryButton } from '../../../components/ui2/buttons/PrimaryButton';
+import { SecondaryButton } from '../../../components/ui2/buttons/SecondaryButton';
+import InputField from '../../../components/ui2/InputField';
+import { Header } from '../../../components/ui2/typography/Header';
+
+export interface EditUserDataProps {
+  block: PasskeyAppendBlock;
+}
+
+export const EditUserData: FC<EditUserDataProps> = ({ block }) => {
+  const { t } = useTranslation('translation', {
+    keyPrefix: `signup.passkey-append.edit-user-data`,
+  });
+  const [passkeyUserHandle, setPasskeyUserHandle] = useState<string>(block.data.userHandle);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (block.data.translatedError) {
+      setLoading(false);
+      setErrorMessage(block.data.translatedError);
+    }
+  }, [block]);
+
+  const headerText = useMemo(() => t('header'), [t]);
+  const primaryButtonText = useMemo(() => t('button_submit'), [t]);
+  const secondaryButtonText = useMemo(() => t('button_cancel'), [t]);
+
+  const handleConfirm = async () => {
+    setLoading(true);
+
+    if (block.data.userHandle === passkeyUserHandle) {
+      block.showPasskeyAppend();
+      return;
+    }
+
+    await block.updateEmail(passkeyUserHandle);
+  };
+
+  return (
+    <div className='new-ui-component'>
```

Comment:
I moved this div and its first child up to AuthFlow.tsx in the error PR, are you fine with this (wanted to have the capability to show global errors on every screen basically)?
If yes, we need to update these new screens (and SignupInit => guess the change goes lost in the merge) now as well.

Reply:
I have not moved them to global space because I don't want to disturb the UI of all the other components. Once all the screens are done I will move this to AuthFlow and will also remove the `new-ui-component` div. If you are okay with  changing the layout of all the screens then I will keep changes in AuthFlow otherwise for now we should only have them in individual components

Reply:
I think it's ok to disturb the old screens and thus make changes to AuthFlow already (the new designs are more important so it's ok that for a few days the old components look a bit messed up on our branch) => I would prefer the second option because then it's easier for Adam and me to give feedback